File: Solution/twoSum.cpp
# twoSum
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> MyMap;
        for(int i = 0; i < nums.size(); i++)
        {
            int diff = target - nums[i];
            if(MyMap.find(diff) != MyMap.end())
            {
                return {MyMap[diff], i};
            }
            MyMap[nums[i]] = i;
        }
        return {};
    }
};
# 求和转换成求差
# 用map结构来存放已经遍历过的数和索引，用map/unordered的find来进行查找，可以减少一层循环
# map结构的用法和map的思想

File: Solution/twoSum.java
// site: https://leetcode-cn.com/problems/two-sum/

class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] result = new int[2];
        Map<Integer, Integer> map = new HashMap<>();
        for(int i = 0; i < nums.length; i++)
        {
            int differ = target - nums[i];
            if(! map.containsKey(differ))
            {
                map.put(nums[i], i);
            }
            else
            {
                result[0] = map.get(differ);
                result[1] = i;
                return result;
            }
        }
        return result;
    }
}
// 用map来实现，map的<K,V>设计是关键，key设计为nums[i]，用differ来查找(map.containsKey)，value为下标
